tests/tools/test_dist_part.py

```diff
@@ -195,25 +149,72 @@ def rand_edges(num_src, num_dst, num_edges):
         with open(meta_fname, 'rb') as f:
             meta_data = json.load(f)
 
-        all_etypes = ['affiliated_with', 'writes', 'cites', 'rev_writes']
         for etype in all_etypes:
             assert len(meta_data['edge_map'][etype]) == num_parts
         assert meta_data['etypes'].keys() == set(all_etypes)
         assert meta_data['graph_name'] == 'mag240m'
 
-        all_ntypes = ['author', 'institution', 'paper']
         for ntype in all_ntypes:
             assert len(meta_data['node_map'][ntype]) == num_parts
         assert meta_data['ntypes'].keys() == set(all_ntypes)
         assert meta_data['num_edges'] == g.num_edges()
         assert meta_data['num_nodes'] == g.num_nodes()
         assert meta_data['num_parts'] == num_parts
 
+        edge_dict = {}
+        edge_data_gold = {}
+
+        if include_edge_data:
+            # Create Id Map here.
+            num_edges = 0
+            for utype, etype, vtype in g.canonical_etypes:
+                fname = ':'.join([utype, etype, vtype])
+                edge_dict[fname] = np.array(
+                    [num_edges, num_edges + g.number_of_edges(etype)]
+                ).reshape(1, 2)
+                num_edges += g.number_of_edges(etype)
+
+            assert num_edges == g.number_of_edges()
+            id_map = dgl.distributed.id_map.IdMap(edge_dict)
+            orig_etype_id, orig_type_eid = id_map(np.arange(num_edges))
+
+            # check edge_data
```

Comment:
as edge data has already been verified in `test_chunk_graph()`, we don't need to verify here. it's duplicate.

Reply:
or below code aims to obtain `edge_data_gold`?

Reply:
You guessed it right. This is just to extract/store edge_data_gold, which will be the ground truth for this test case.